### PR TITLE
e2e: prohibited using helm, use azure.json

### PR DESCRIPTION
**Pull request by akshaysngupta**
## Checklist
- [x] The title of the PR is clear and informative
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] Issues addressed by the PR are mentioned in the description followed by `Fixes`.

## Description

This PR adds e2e for 2 cases:
1. Use the new helm config way of adding prohibited target
2. Make sure that AGIC is able to use azure.json by just providing gateway name and no other information. AGIC should get remaining info from azure.json.

### CODE REVIEW

File: scripts/e2e/run-e2e.sh [hunk 1]
```diff
@@ -47,28 +50,24 @@ function InstallAGIC() {
...
-    # apply customized prohibited policy for blacklisted backend
-    kubectl apply -f prohibit-blacklist-service.yaml
-
     # get all the prohibited target config
-    kubectl get AzureIngressProhibitedTargets -n agic -o yaml 
-
-    # delete default prohibit-all-targets
-    # blacklist-service shall be kept after porhibited policy applied
-    kubectl delete AzureIngressProhibitedTarget prohibit-all-targets -n agic
+    kubectl get AzureIngressProhibitedTargets -n agic -o yaml
```
Comment (3quanfeng) [inline scripts/e2e/run-e2e.sh:70]: why removed  the line of "kubectl delete AzureIngressProhibitedTarget prohibit-all-targets -n agic"
the broader policy by default should be deleted

File: scripts/e2e/run-e2e.sh [hunk 2]
```diff
@@ -12,10 +12,12 @@ function DeleteOtherAGICVersions() {
 }
 
 function InstallAGIC() {
+    # Using 'applicationGatewayName' without providing 'subscription' and 'resource group' will make AGIC use values from azure.json
     [[ -z "${version}" ]] && (echo "version is not set"; exit 1)
-    [[ -z "${applicationGatewayId}" ]] && (echo "buiapplicationGatewayIdldid is not set"; exit 1)
+    [[ -z "${applicationGatewayName}" ]] && (echo "applicationGatewayName is not set"; exit 1)
     [[ -z "${identityResourceId}" ]] && (echo "identityResourceId is not set"; exit 1)
     [[ -z "${identityClientId}" ]] && (echo "identityClientId is not set"; exit 1)
+    [[ -z "${applicationGatewaySubnetPrefix}" ]] && (echo "applicationGatewaySubnetPrefix is not set"; exit 1)
```
Comment (3quanfeng) [inline scripts/e2e/run-e2e.sh:20]: why it's mandatory for install(exit 1)?

File: scripts/e2e/run-e2e.sh (hunk 1 above)
Comment (akshaysngupta) [reply scripts/e2e/run-e2e.sh:70]: this policy is not installed when using prohibited target in helm.

File: scripts/e2e/run-e2e.sh (hunk 2 above)
Comment (akshaysngupta) [reply scripts/e2e/run-e2e.sh:20]: right now, this mandatory in helm when using appgw name instead of appgw id.

**Review by 3quanfeng — APPROVED** [review]

**Pull request merged**